fuel-relayer/src/validators.rs

```diff
@@ -26,16 +26,94 @@ impl Validators {
         self.set = db.get_validators().await;
     }
 
-    /// Get validator set
+    /// Get validator set for da_height.
     pub async fn get(
-        &mut self,
+        &self,
         da_height: DaBlockHeight,
+        db: &mut dyn RelayerDb,
     ) -> Option<HashMap<Address, (u64, Option<Address>)>> {
-        // TODO apply down drift https://github.com/FuelLabs/fuel-core/issues/365
-        if self.da_height == da_height {
-            return Some(self.set.clone());
+        match self.da_height.cmp(&da_height) {
+            std::cmp::Ordering::Less => {
+                // We request validator set that we still didnt finalized or know about.
+                // Probably there is eth client sync problem
+                error!(
+                    "current height {} is less then requested validator set height: {da_height}",
+                    self.da_height
+                );
+                return None;
+            }
+            std::cmp::Ordering::Equal => {
+                // unusual but do nothing
+                info!("Get last finalized set height: {da_height}");
+                let set = self
+                    .set
+                    .iter()
+                    .filter(|(_, (stake, consensus_key))| *stake != 0 && consensus_key.is_some())
+                    .map(|(k, v)| (*k, *v))
+                    .collect();
+                return Some(set);
+            }
+            std::cmp::Ordering::Greater => {
+                // slightly drift to past
+            }
+        }
+
+        let mut validators = self.set.clone();
+        // get staking diffs to revert from our current set.
+        let diffs = db
+            .get_staking_diffs(da_height + 1, Some(self.da_height))
```

Comment:
Should we use `checked_add` to prevent overflow from a user input causing use to diff way more history than expected?